code-samples/creating-your-bot/event-handling/events/guildMemberUpdate.js

```diff
@@ -1,21 +1,48 @@
 module.exports = {
 	name: "guildMemberUpdate",
 	async execute(oldMember, newMember) {
-		console.log(oldMember, newMember);
-		// if (!interaction.isChatInputCommand()) return;
+		//console.log(oldMember, newMember);
+    
+    //var defaultHost = "https://forum.citydao.io/";
+    
+    /* 
+    / Grab object for the new role
+    / newRoleId = ID of the changed Role
+    / newRoleName = Name of the changed Role
+    */
+    var newRole = newMember.roles.cache.difference(oldMember.roles.cache).last();
+    var newRoleId = newRole.id;
+    var newRoleName = newRole.name; 
+		//console.log(newRole);
+		console.log(`Role ID: ${newRoleId}`);
+		console.log(`Role Name: ${newRoleName}`);
 
-		// const command = interaction.client.commands.get(interaction.commandName);
+		console.log(newMember.roles.cache);
 
-		// if (!command) {
-		// 	console.error(`No command matching ${interaction.commandName} was found.`);
-		// 	return;
-		// }
+    /* 
+    / MemberNickName = Discord nickname of the current user (string)
+    / MemberId = Discord ID of the current user (snowflake)
+    / MemberName = Discord username (if no username then nickname) of the current user (string)
+    */ 
+    var MemberNickName = newMember.nickname;
+    var MemberId = newMember.id;
+    var MemberName = newMember.displayName;
+    console.log(`Nickname: ${MemberNickName}`);
+    console.log(`ID: ${MemberId}`);
+    console.log(`Name: ${MemberName}`);
 
-		// try {
-		// 	await command.execute(interaction);
-		// } catch (error) {
-		// 	console.error(`Error executing ${interaction.commandName}`);
-		// 	console.error(error);
-		// }
+    // check whether we added or removed roles
+    newMemberSize = newMember.roles.cache.size;
+    oldMemberSize = oldMember.roles.cache.size;
+		console.log(`new size: ${newMemberSize}`);
+		console.log(`old size: ${oldMemberSize}`);
+		// needed to +1 for oldMemberSize due to @everyone role in discord
+    if(newMemberSize > oldMemberSize+1) {
```

Comment:
I don't understand this logic? Why are we looking at size?

Reply:
If the new size is greater than old one, then we know that the role was added to the user.
If the new size is less than the old one, then we know that the role was removed from the user.

Reply:
This still won't tell you what role was added or removed?

Reply:
We find that here
https://github.com/Slyracoon23/DiscordBot-Discourse-Sync/blob/63868d8b310d40a126b489d191af4fd7bc8d07a3/code-samples/creating-your-bot/event-handling/events/guildMemberUpdate.js#L18